pkgs/development/libraries/libtheora/default.nix

```diff
@@ -20,6 +20,10 @@ stdenv.mkDerivation rec {
     substituteInPlace configure --replace "-fforce-addr" ""
   '';
 
+  # libtheora's configure script is generated with an old version of libtool.
+  # Regenerate with a modern version:
+  preConfigure = "sh autogen.sh";
+
```

Comment:
```suggestion
```

autoreconfHook should take care of this.

Reply:
Autoconf is starting to make me feel very dumb, I don't think I can fix it. Made your suggested changes but run into errors like:

```
libtheora> unpacking sources
libtheora> unpacking source archive /nix/store/1s6ssdh3sjciqc0lzny58bf0iymbadzy-libtheora-1.1.1.tar.gz
libtheora> source root is libtheora-1.1.1
libtheora> setting SOURCE_DATE_EPOCH to timestamp 1254420253 of file libtheora-1.1.1/config.h.in
libtheora> patching sources
libtheora> autoreconfPhase
libtheora> autoreconf: export WARNINGS=
libtheora> autoreconf: Entering directory '.'
libtheora> autoreconf: configure.ac: not using Gettext
libtheora> autoreconf: running: aclocal --force
libtheora> configure.ac:316: warning: macro 'AM_PATH_SDL' not found in library
libtheora> autoreconf: configure.ac: tracing
libtheora> autoreconf: running: libtoolize --copy --force
libtheora> libtoolize: putting auxiliary files in '.'.
libtheora> libtoolize: copying file './ltmain.sh'
libtheora> libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
libtheora> libtoolize: and rerunning libtoolize and aclocal.
libtheora> libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
libtheora> autoreconf: configure.ac: not using Intltool
libtheora> autoreconf: configure.ac: not using Gtkdoc
libtheora> autoreconf: running: aclocal --force
libtheora> configure.ac:316: warning: macro 'AM_PATH_SDL' not found in library
libtheora> autoreconf: running: /nix/store/1jx7wq4c3d93v585jzlyra0hn99djk6i-autoconf-2.70/bin/autoconf --force
libtheora> configure.ac:12: warning: 'AM_CONFIG_HEADER': this macro is obsolete.
libtheora> configure.ac:12: You should use the 'AC_CONFIG_HEADERS' macro instead.
libtheora> ./lib/autoconf/general.m4:2433: AC_DIAGNOSE is expanded from...
libtheora> aclocal.m4:10322: AM_CONFIG_HEADER is expanded from...
libtheora> configure.ac:12: the top level
libtheora> configure.ac:60: warning: The macro `AC_LIBTOOL_WIN32_DLL' is obsolete.
libtheora> configure.ac:60: You should run autoupdate.
libtheora> aclocal.m4:8769: AC_LIBTOOL_WIN32_DLL is expanded from...
libtheora> configure.ac:60: the top level
libtheora> configure.ac:60: warning: AC_LIBTOOL_WIN32_DLL: Remove this warning and the call to _LT_SET_OPTION when you
libtheora> configure.ac:60: put the 'win32-dll' option into LT_INIT's first parameter.
libtheora> ./lib/autoconf/general.m4:2433: AC_DIAGNOSE is expanded from...
libtheora> aclocal.m4:8769: AC_LIBTOOL_WIN32_DLL is expanded from...
libtheora> configure.ac:60: the top level
libtheora> configure.ac:61: warning: The macro `AM_PROG_LIBTOOL' is obsolete.
libtheora> configure.ac:61: You should run autoupdate.
libtheora> aclocal.m4:376: AM_PROG_LIBTOOL is expanded from...
libtheora> configure.ac:61: the top level
libtheora> configure.ac:208: warning: The macro `AC_PROG_LD' is obsolete.
libtheora> configure.ac:208: You should run autoupdate.
libtheora> aclocal.m4:3598: AC_PROG_LD is expanded from...
libtheora> configure.ac:208: the top level
libtheora> configure.ac:276: warning: The macro `AC_HELP_STRING' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:276: warning: The macro `AC_TRY_RUN' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2996: AC_TRY_RUN is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:276: warning: The macro `AC_TRY_LINK' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:310: warning: The macro `AC_HELP_STRING' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:310: warning: The macro `AC_TRY_RUN' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2996: AC_TRY_RUN is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:310: warning: The macro `AC_TRY_LINK' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:452: warning: AC_OUTPUT should be used without arguments.
libtheora> configure.ac:452: You should run autoupdate.
libtheora> configure.ac:316: error: possibly undefined macro: AM_PATH_SDL
libtheora> If this token and others are legitimate, please use m4_pattern_allow.
libtheora> See the Autoconf documentation.
libtheora> configure.ac:469: error: possibly undefined macro: AS_AC_EXPAND
libtheora> autoreconf: error: /nix/store/1jx7wq4c3d93v585jzlyra0hn99djk6i-autoconf-2.70/bin/autoconf failed with exit status: 1
```

Reply:
I've fixed this with SuperSandro2000's help. Will submit a new patch soon...